Binding svc to sd-jwt (Issue #31) (#34)
oauthstuff/draft-selective-disclosure-jwt: main.md

Note: need to define how holder public key is included, right now examples are using `sub_jwk` I think.

#### Hash of an SVC

* `svc_hash`

>   * REQUIRED. SVC hash value. Its value is the base64url encoding of the left-most half of the hash of the octets of the ASCII representation of the SVC value, where the hash algorithm used is the same hash algorithm that is used for the selectively disclosable claims in an SD-JWT. The svc_hash value is a case sensitive string.

Reviewer: I have questions about "the ASCII representation of the SVC value".  Is this representation the entire SVC, such as in the example at https://www.ietf.org/archive/id/draft-fett-oauth-selective-disclosure-jwt-00.html#section-4.2.2 ?  If so, ASCII should almost certainly be replaced by UTF-8, since JSON can contain non-ASCII characters.  If it's something else, then we also likely need to be more descriptive about what it is.

Reviewer 2: UTF-8 contains ASCII, this latter is a subset of the first.
The hashes are hexadecimal values in ascii format.

We may say that the text is good as it is if I'm not missing something @selfissued :)

For transporting the SVC together with the SD-JWT from the issuer to the holder,
the SVC is base64url-encoded and appended to the SD-JWT using a period character `.` as the
separator. 

The SVC and SD-JWT are implicitly linked through the hash values of the claims
in the SVC that is included in the SD-JWT. To ensure that the correct SVC and 

> SD-JWT pairings are being used, the holder should verify the binding between

Reviewer 3: ```suggestion
SD-JWT pairings are being used, the holder SHOULD verify the binding between
```

> SVC and SD-JWT, as defined in the Verification Section of this document.

Reviewer 3: I think this sentence works with and without the comma, but without the comma is probably closer to what we want to say.

```suggestion
SVC and SD-JWT as defined in the Verification Section of this document.
```

For transporting the SVC together with the SD-JWT from the issuer to the holder,
the SVC is base64url-encoded and appended to the SD-JWT using a period character `.` as the
separator. 

> The SVC and SD-JWT are implicitly linked through the hash values of the claims
> in the SVC that is included in the SD-JWT. To ensure that the correct SVC and 
> SD-JWT pairings are being used, the holder should verify that all the claims 
> in the SVC are present in the SD-JWT, that there are no claims in the SD-JWT 
> that are not in the SVC and that the hashes of the claims in the SVC match 
> those in the SD-JWT.

Author: ```suggestion
The SVC and SD-JWT are implicitly linked through the hash values of the claims
in the SVC that is included in the SD-JWT. To ensure that the correct SVC and 
SD-JWT pairings are being used, the holder should verify the binding between
SVC and SD-JWT, as defined in the Verification Section of this document.
```

# Verification

## Verification by the Holder when Receiving SD-JWT and SVC

> The holder SHOULD verify the binding between SD-JWT and SVC by following the following steps:

Author: ```suggestion
The holder SHOULD verify the binding between SD-JWT and SVC by performing the following steps:
```

 1. Check that all the claims in the SVC are present in the SD-JWT and that there are no claims in the SD-JWT that are not in the SVC

>  2. Check and that the hashes of the claims in the SVC match those in the SD-JWT

Reviewer 4: Editorial 0 delete "and"

Check and that the hashes of the claims in the SVC match those in the SD-JWT

>  2. Check and that the hashes of the claims in the SVC match those in the SD-JWT

Author: ```suggestion
 2. Check that the hashes of the claims in the SVC match those in the SD-JWT
```
